File: python_back/xpander_sdk/models/payloads.py
from pydantic import BaseModel
from typing import Any

class RequestPayload(BaseModel):
    """
    A model to represent the request payload.

    Attributes:
        body_params (Any): The body parameters of the request.
        query_params (Any): The query parameters of the request.
        path_params (Any): The path parameters of the request.
        headers (Any): The headers of the request.
    """
    body_params: Any
    query_params: Any
    path_params: Any
    headers: Any

File: src/models/payloads.ts
// models/payloads.ts

/**
 * Class representing the request payload.
 */
export class RequestPayload {
  /**
   * The body parameters of the request.
   */
  body_params: any;

  /**
   * The query parameters of the request.
   */
  query_params: any;

  /**
   * The path parameters of the request.
   */
  path_params: any;

  /**
   * The headers of the request.
   */
  headers: any;

  /**
   * Constructs a new RequestPayload instance.
   * @param body_params - The body parameters of the request.
   * @param query_params - The query parameters of the request.
   * @param path_params - The path parameters of the request.
   * @param headers - The headers of the request.
   */
  constructor(
    body_params: any,
    query_params: any,
    path_params: any,
    headers: any,
  ) {
    this.body_params = body_params;
    this.query_params = query_params;
    this.path_params = path_params;
    this.headers = headers;
  }
}
